api/src/wfl/module/covid.clj

```diff
@@ -103,8 +103,76 @@
   "Use `tx` to load the workload sink with `sink_type`."
   (fn [tx workload] (:sink_type workload)))
 
+;; validations
+(defmulti throw-when-malformed-source-request!
+  "Validate the source of a `request`"
+  (fn [source] (:name source)))
+
+(defmulti throw-when-malformed-executor-request!
+  "Validate the executor of a `request`"
+  (fn [executor] (:name executor)))
+
+(defmulti throw-when-malformed-sink-request!
+  "Validate the sink of a `request`"
+  (fn [sink] (:name sink)))
+
+;; :default implementations
+(defmethod throw-when-malformed-source-request!
+  :default
+  [source]
+  (throw
+   (ex-info "Failed to create workload - unknown source"
+            {:source source
+             :type  ::invalid-source})))
+
+(defmethod throw-when-malformed-executor-request!
+  :default
+  [executor]
+  (throw
+   (ex-info "Failed to create workload - unknown executor"
+            {:source executor
+             :type  ::invalid-executor})))
+
+(defmethod throw-when-malformed-sink-request!
+  :default
+  [sink]
+  (throw
+   (ex-info "Failed to create workload - unknown sink"
+            {:source sink
+             :type  ::invalid-sink})))
+
+
+;; Workload Functions
```

Comment:
nit: fix the dangling comment

Reply:
Blank lines nit: see [Clojure style guide](https://github.com/bbatsov/clojure-style-guide#empty-lines-between-top-level-forms).

Reply:
Done